rfcs/0168-feature-parameter-names.md

```diff
@@ -0,0 +1,559 @@
+  for the purposes of consistency with other packages.
+* Further contributions that improve compliance with this RFC MAY involve
+  maintainers but SHOULD NOT be required to do so because the rules SHOULD be
+  clear.
+* New packages MUST be compliant with the RFC (either no feature parameters or compliant with this RFC.)
+
+This ensures that count of non-compliant feature parameters in nixpkgs is
+non-increasing over the time.
+
+### Technical process
+
+Named parameters are part of the programming interface, and renaming them is a
+breaking change. As such, renaming of the parameters is done in following way:
+
+1. Following function is added into `lib` set:
+
+```
+let renamed = { oldName, newName, sunset, oldValue }: newValue:
+   let warning = builtins.concatStringsSep " "
+      [ "Feature flag"
+      , oldName
+      , "is renamed to"
+      , newName
+      , "; old name will no longer be available in nixpkgs="
+      , sunset
+      , "."
+      ]
+   in lib.warnIf (value != null) warning (if (value != null) then value else newValue);
+```
+
+Starting with following package:
+```
+{ lib
+, stdenv
+, nonCompliantFoo ? true
+}:
+
+# uses nonCompliantFoo
+stdenv.mkDerivation { ... }
+```
+
+the first step of migration is to replace it with the following:
+```
+{ lib
+, stdenv
+, nonCompliantFoo ? null
+, enable_foo ? lib.renamed {
+    oldName = "nonCompliantFoo";
+    newName = "enable_foo";
+    sunset = "25.11";
+    value = nonCompliantFoo;
+  } true
+}:
+
+# uses enable_foo
+stdenv.mkDerivation { ... }
+```
+
+where `sunset` parameter is set two full releases after time of this change.
+So, if this change is made in Jan 2024, `sunset` MUST be set to `25.11`. After
+the release `25.05` all mentions of `nonCompliantFoo` SHOULD be removed, and
+package will look like following:
+
+```
+{ lib
+, stdenv
+, enable_foo ? true
+}:
+
+# uses enable_foo
+stdenv.mkDerivation { ... }
+```
+
+## Feature parameter naming guidelines
+
+Rules in the first section of this RFC describe different scenarios when
+`with`, `enable` and `conf` feature parameters must be used, but coming with
+exact name that is clear, concise, describes it effect correctly and is
+consistent across the nixpkgs is hard to codify with absolute rule. Instead,
+this RFC provides set of guidelines and examples based on scenarios brought up
+during the discussion.
+
+Feature flags that require single external dependency SHOULD be named after
+that dependency. Prefix `lib` SHOULD be removed. For example,
+
+<details><summary>Example</summary>
+
+```
+systemd => with_systemd
+libgif  => with_gif
+curl    => with_curl
+alsa    => with_alsa
+
+# When people say "Xorg" support, they usually mean linking "X11"
+# library which is one of the ways to talk X protocol to X server. "xcb"
+# being another one.
+#
+# I am not aware of anybody implementing X protocol themself, and not
+# using one of these libraries.
```

Comment:
`sbcl.pkgs.clx` does, but I am not aware of any Common Lisp package with an optional dependency on CLX. Rust has e.g. `breadx`.